blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/BasicFieldVariableView.java

```diff
@@ -225,32 +229,31 @@ public void onChanged() {
          * @return The index of the variable.
          */
         public int getOrCreateVariableIndex(String variableName) {
-            int count = mVariableNameManager.size();
-            for (int i = 0; i < count; i++) {
-                if (variableName.equalsIgnoreCase(getItem(i))) {
-                    return i;
-                }
+            String existing = mVariableNameManager.getExisting(variableName);
+            if (existing != null) {
+                return getIndexForVarName(existing);
+            } else {
+                // No match found.  Create it.
+                variableName = mVariableNameManager.makeValidName(
+                        /* suggested */ variableName, /* fallback */ variableName);
+                mVariableNameManager.addName(variableName);
+                int insertionIndex = getIndexForVarName(variableName);
+
+                notifyDataSetChanged();
+                return insertionIndex;
             }
+        }
 
-            // No match found.  Create it.
-            mVariableNameManager.addName(variableName);
-
-            // Reindex, finding the new index along the way.
-            count = mVariableNameManager.size();
-            clear();
-            int insertionIndex = -1;
-            for (int i = 0; i < count; i++) {
-                add(mVariableNameManager.get(i));
-                if (variableName.equals(getItem(i))) {
-                    insertionIndex = i;
+        private int getIndexForVarName(String expected) {
+            int i = 0;
+            for (String varName : mVars) {
```

Comment:
You've been converting a lot of direct access loops to iterator loops. These are slightly slower and cause a bit more garbage collection. It's not significant, but I'd prefer you not convert them unless there's a reason to.